### PR TITLE
CTW-381 Autofill Values for Confirming a Condition

### PR DESCRIPTION
**Pull request by melocule**
Sets verification status to active by default when user adds a condition from another provider, not just when the user adds a new condition. 
Adds a helper function to set any common default values between adding a new condition or confirming one.

https://user-images.githubusercontent.com/33408946/197026210-1667abd3-4097-458f-8425-85b15a59be78.mov

### CODE REVIEW

**Comment by luthfib** [comment]
I think your branch has some funky stuff going on I dont think you changed 26 files

**Comment by pmadria3** [comment]
Something weird happened with the merge (did you rebase instead of merge potentially?)

File: src/components/content/conditions.tsx [hunk 1]
```diff
@@ -86,15 +87,21 @@ export function Conditions({ className }: ConditionsProps) {
     if (patientResponse.data) {
       setDrawerIsOpen(true);
       setFormAction("Edit");
-      setCurrentlySelectedData(getEditingPatientConditionData({ condition }));
+      setCurrentlySelectedData(getEditConditionFormEntries({ condition }));
     }
   };
 
-  const handleOtherProviderRecordsCondition = (condition: ConditionModel) => {
+  const handleConfirmCondition = (condition: ConditionModel) => {
```
Comment (luthfib) [inline src/components/content/conditions.tsx:94]: Little confused by this rename? Not too sure `handleConfirmCondition` is descriptive of whats going on

File: src/components/content/forms/condition-helpers.ts [hunk 2]
```diff
@@ -87,3 +92,31 @@ const sharedFields = (condition: ConditionModel) => [
     field: "note",
   },
 ];
+
+// Sets any autofill values that apply when a user adds a medication, whether creating or confirming.
```
Comment (luthfib) [inline src/components/content/forms/condition-helpers.ts:96]: You say medication here, I think you mean condition

**Review by luthfib — COMMENTED** [review]
Looking good, some minor nitpicks

File: src/fhir/conditions.ts [hunk 3]
```diff
@@ -52,33 +51,14 @@ export type QueryKeyPatientConditions = [
...
-        {
-          system: SYSTEM_CONDITION_VERIFICATION_STATUS,
-          code: "confirmed",
-          display: "Confirmed",
-        },
-      ],
-      text: "confirmed",
-    },
   };
+  newCondition = applyAddConditionDefaults(newCondition);
```
Comment (WesleyKapow) [inline src/fhir/conditions.ts:61]: NOTE: `applyAddConditionDefaults` modifies `newCondition` argument, so this assignment is not needed.

File: src/components/content/forms/condition-helpers.ts [hunk 4]
```diff
@@ -87,3 +92,31 @@ const sharedFields = (condition: ConditionModel) => [
...
+          system: SYSTEM_CONDITION_VERIFICATION_STATUS,
+          code: "confirmed",
+          display: "Confirmed",
+        },
+      ],
+      text: "confirmed",
+    },
+  };
+
+  return Object.assign(condition, addDefaults);
```
Comment (WesleyKapow) [inline src/components/content/forms/condition-helpers.ts:121]: No need to return as you are modifying the passed in condition via this `assign`

**Review by WesleyKapow — CHANGES_REQUESTED** [review]

**Review by WesleyKapow — DISMISSED** [review]

File: src/components/content/forms/conditions-autocomplete.tsx [hunk 5]
```diff
@@ -1,11 +1,44 @@
...
+} from "@/fhir/system-urls";
+import { Condition } from "fhir/r4";
 import { InputHTMLAttributes, useState } from "react";
 import {
   ComboboxField,
   ComboxboxFieldOption,
 } from "../../core/form/combobox-field";
 
+// Sets any autofill values that apply when a user adds a condition, whether creating or confirming.
+export function setAddConditionDefaults(condition: Condition): void {
```
Comment (WesleyKapow) [inline src/components/content/forms/conditions-autocomplete.tsx:15]: This has nothing to do with autcomplete, move somewhere else. Maybe fhir/conditions?

**Review by WesleyKapow — DISMISSED** [review]

File: src/components/content/forms/conditions-autocomplete.tsx (hunk 5 above)
Comment (melocule) [reply src/components/content/forms/conditions-autocomplete.tsx]: Moved to the file with condition helpers for forms

**Review by WesleyKapow — APPROVED** [review]
Approving but in general our `fhir/` code should not depend on our form code. Should always be the other way around as the `fhir/` code is more "basic" or a "lower level".

**Pull request merged**